core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -298,6 +305,7 @@ public NodeIndicesStats stats(boolean includePrevious, CommonStatsFlags flags) {
     /**
      * Returns <tt>true</tt> if changes (adding / removing) indices, shards and so on are allowed.
      */
+    @Override
     public boolean changesAllowed() {
```

Comment:
yes, we also stop `ClusterService` (that is causing the cluster change events) before we stop `IndicesService` so we know that `IndicesClusterStateService` cannot create new indices based on cluster change events anymore after we are cleaning up in `IndicesService`.

In theory, no checks would be needed at all.

In practice, I wonder if it's good enough to guard `createIndex` (already the case) and `createShard` in `IndicesService` with
```
if (!lifecycle.started()) {
    throw new IllegalStateException(...);
}
```

and just remove the checks in `IndicesClusterStateService`.